Quality_Declaration_foonathan_memory.md

```diff
@@ -0,0 +1,205 @@
+If ROS 2 dependencies are found, `foonathan_memory_vendor` could run ROS 2 linters and copyright tests as can be seen in the [nightly CI results](https://ci.ros2.org/view/nightly/job/nightly_linux_release/lastBuild/testReport/foonathan_memory_vendor/).
+
+### Documentation Policy [2.v]
+
+`foonathan_memory` does not have a stated documentation policy.
+
+## Documentation [3]
+
+### Feature Documentation [3.i]
+
+`foonathan_memory` provides a [feature list](https://foonathan.net/memory/index.html) with descriptions of its main features.
+
+### Public API Documentation [3.ii]
+
+`foonathan_memory` has embedded API documentation that is generated using [Doxygen](https://www.doxygen.nl/index.html) and is [hosted](https://foonathan.net/memory/namespacefoonathan_1_1memory.html) alongside the feature documentation.
+
+Additionally, *eprosima Fast DDS* provides a [document](Public_API_foonathan_memory.md) stating the API used within the project.
+
+### License [3.iii]
+
+The license for `foonathan_memory` is Zlib.
+A summary statement is provided in each source file and a full copy can be found in the [LICENSE](https://raw.githubusercontent.com/foonathan/memory/master/LICENSE) file.
+
+### Copyright Statements [3.iv]
+
+The copyright holders each provide a statement of copyright in each source file in `foonathan_memory`.
+
+## Testing [4]
+
+### Feature and Public API Testing [4.i]/[4.ii]
+
+`foonathan_memory` provides tests which simulate typical usage located in the [`test`](https://github.com/foonathan/memory/tree/master/test) directory.
+
+Specifically, the API used by *eprosima Fast DDS* is tested in the following tests:
+
+* Allocators: [test/default_allocator.cpp](https://github.com/foonathan/memory/blob/c619113a616c5771ee693c7abdcef284e02f7d61/test/default_allocator.cpp)
+* Allocator implementations: [test/memory_pool.cpp](https://github.com/foonathan/memory/blob/c619113a616c5771ee693c7abdcef284e02f7d61/test/memory_pool.cpp)
+* Adapters and Wrappers: [test/segregator.cpp](https://github.com/foonathan/memory/blob/c619113a616c5771ee693c7abdcef284e02f7d61/test/segregator.cpp)
+* Alias templates: `foonathan_memory` does not provide any tests to check this functionality.
+Regardless, *eprosima Fast DDS* tests these features in the [PersistenceTests](https://github.com/eProsima/Fast-DDS/tree/master/test/unittest/rtps/persistence) and the [WriterProxyTests](https://github.com/eProsima/Fast-DDS/tree/master/test/unittest/rtps/reader).
+
+### Coverage [4.iii]
+
+`foonathan_memory` does not track testing coverage.
+Anyway, *eprosima Fast DDS* ensures that every feature and API used within the library has been tested.
+
+### Performance [4.iv]
+
+`foonathan_memory` does not conduct performance tests.
+
+*eprosima Fast DDS* will run its [performance tests](test/performance) and analyzed their results before deciding to change the pinned commit to prevent performance regression.
```

Comment:
```suggestion
*eprosima Fast DDS* will run its [performance tests](test/performance) and analyze their results before deciding to change the pinned commit to prevent performance regression.
```